src/audio/module_adapter/module_adapter.c

```diff
@@ -55,7 +55,7 @@ struct comp_dev *module_adapter_new(const struct comp_driver *drv,
 	dev->ipc_config = *config;
 	dev->drv = drv;
 
-	mod = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM, sizeof(*mod));
+	mod = rzalloc(SOF_MEM_ZONE_RUNTIME_SHARED, 0, SOF_MEM_CAPS_RAM, sizeof(*mod));
```

Comment:
Please use kconfig option - non_coherent_pipeline_workaround or similar
this is not a target sollution, not a fix at all

Reply:
how can a buffer and a module, attached to it, be allocated on different cores? Shouldn't both be done on the target core, where the pipeline will run? However, if this is for the case when parts of a stream are running on different cores, then this is still not supported and we still have bigger issues to fix for that. It is good to gradually prepare for that and fix parts that we come across, but I'd agree with @marcinszkudlinski that this needs to be considered carefully: ATM we don't have such use cases so for now it doesn't fix anything, but it probably will slow things down somewhat.

Reply:
That ipc4_create_buffer() was called for buffer between two components on different pipelines and pipelines are on different cores. To get obs size ipc4_create_buffer() reaches component's base_cfg. The problem is that ipc4_create_buffer() was called from one core but that componet's base_cfg was allocated and written on another core.

Seems such problem only happens for buffers between pipelines on different cores, not sure.

So, yes, probably, this problem should be fixed by ensuring ipc4_create_buffer() is called from correct core.

Reply:
@serhiy-katsyuba-intel connecting components or pipelines on different cores is unsupported ATM

Reply:
we need to fix it, cross core bind LL-DP is something we DO need and we need it soon

Reply:
@marcinszkudlinski sure, so somebody has to get to work on this and get it done ASAP